effekt/shared/src/main/scala/effekt/core/Transformer.scala

```diff
@@ -170,10 +170,45 @@ object Transformer extends Phase[Typechecked, CoreTransformed] {
     Box(transformAsBlock(tree), transform(Context.inferredCapture(tree)))
 
   def transformAsBlock(tree: source.Term)(using Context): Block = tree match {
```

Comment:
I think this will translate too many cases. For example in the call `println(42)` it might eta expand `println`, no?

Reply:
Maybe we can have a separate `transformAsControlBlock` or similar (ideally with a better name / comment what that means)?

Reply:
I had a `transformAsBlockArgument` originally, but then noticed that there were no relevant usages of the original function left.
In function calls, the callee is always an Id: https://github.com/effekt-lang/effekt/blob/d4a12916ae5b38326b4dbc30aa72ae90e0d88e36/effekt/shared/src/main/scala/effekt/core/Transformer.scala#L407
Called from here, directly with source symbol: https://github.com/effekt-lang/effekt/blob/d4a12916ae5b38326b4dbc30aa72ae90e0d88e36/effekt/shared/src/main/scala/effekt/core/Transformer.scala#L336-L338
Receivers of method calls get fed through this, but are not wrapped, so this should be fine IIUC.

Reply:
Usages of `transformAsBlock` are:
- https://github.com/effekt-lang/effekt/blob/d4a12916ae5b38326b4dbc30aa72ae90e0d88e36/effekt/shared/src/main/scala/effekt/core/Transformer.scala#L66
  https://github.com/effekt-lang/effekt/blob/d4a12916ae5b38326b4dbc30aa72ae90e0d88e36/effekt/shared/src/main/scala/effekt/core/Transformer.scala#L149
  => must be wrapped, since any symbol referring to a definition is always called using `App`
- https://github.com/effekt-lang/effekt/blob/d4a12916ae5b38326b4dbc30aa72ae90e0d88e36/effekt/shared/src/main/scala/effekt/core/Transformer.scala#L170
  => I thought boxed functions must be wrapped. Apparently, this is not necessarily the case. However I suspect this might be a problem:
  https://github.com/effekt-lang/effekt/blob/d4a12916ae5b38326b4dbc30aa72ae90e0d88e36/effekt/shared/src/main/scala/effekt/core/Transformer.scala#L330-L334
  If we `box` a toplevel definition this will lead to the original issue IIUC :thinking:
- https://github.com/effekt-lang/effekt/blob/d4a12916ae5b38326b4dbc30aa72ae90e0d88e36/effekt/shared/src/main/scala/effekt/core/Transformer.scala#L178
  https://github.com/effekt-lang/effekt/blob/d4a12916ae5b38326b4dbc30aa72ae90e0d88e36/effekt/shared/src/main/scala/effekt/core/Transformer.scala#L295-L296
  => here, the block is an interface, so no wrapping will occur
- https://github.com/effekt-lang/effekt/blob/d4a12916ae5b38326b4dbc30aa72ae90e0d88e36/effekt/shared/src/main/scala/effekt/core/Transformer.scala#L299
  https://github.com/effekt-lang/effekt/blob/d4a12916ae5b38326b4dbc30aa72ae90e0d88e36/effekt/shared/src/main/scala/effekt/core/Transformer.scala#L327
  https://github.com/effekt-lang/effekt/blob/d4a12916ae5b38326b4dbc30aa72ae90e0d88e36/effekt/shared/src/main/scala/effekt/core/Transformer.scala#L413
  => These are block arguments, must be wrapped.

Reply:
I could still rename the function to make this more obvious.

Reply:
Actually, for the "is an interface" cases, we might want to have a separate `transformAsInterfaceBlock`? This also adds an additional sanity check, that we don't call methods on a function or try to call an interface (instead of one of it's methods). 🤔

Reply:
> we might want to have a separate transformAsInterfaceBlock?
I think this is fine not to have :)

Reply:
After having done that split, I am not entirely sure it is a good idea: There is no case where we call `transformAsControlBlock` from the rest of the transformation (as there are only symbols in function position), and it adds some duplication between the two (esp. Select).
Also, dispatching is more convoluted than it should be IMHO.

Reply:
I see you concerns. At the same time, I like this because we don't pretend to automagically eta-expand objects :) Not sure... hmm...

Reply:
That's true. I'll leave it as it is.